velox/functions/prestosql/aggregates/tests/PrestoHasherTest.cpp

```diff
@@ -0,0 +1,343 @@
+      PrestoHasher hasher(*vector, rows);
+
+      BufferPtr hashes =
+          AlignedBuffer::allocate<int64_t>(vectorSize, pool_.get());
+      hasher.hash(rows, hashes);
+
+      auto rawHashes = hashes->as<int64_t>();
+      for (int i = 0; i < expected.size(); i++) {
+        EXPECT_EQ(expected[i], rawHashes[i]) << "at " << i;
+      }
+    };
+
+    checkHashes(vector, expected);
+
+    // Wrap in Dictionary
+    BufferPtr indices =
+        AlignedBuffer::allocate<vector_size_t>(vectorSize, pool_.get());
+    auto rawIndices = indices->asMutable<vector_size_t>();
+    // Just 1:1 mapping.
+    for (size_t i = 0; i < vectorSize; ++i) {
+      rawIndices[i] = i;
+    }
+    auto dictionaryVector = BaseVector::wrapInDictionary(
+        BufferPtr(nullptr), indices, vectorSize, vector);
+
+    checkHashes(dictionaryVector, expected);
+
+    // Wrap in constant.
+    std::vector<int64_t> constantExpected{0};
+    for (int i = 0; i < vectorSize; i++) {
+      constantExpected[0] = expected[i];
+      auto constantVector = BaseVector::wrapInConstant(1, i, vector);
+      checkHashes(constantVector, constantExpected);
+    }
+  }
+
+  template <typename T>
+  void testIntegral() {
+    assertHash<T>(
+        {1, 0, 100, std::nullopt},
+        {9155312661752487122, 0, -367469560765523433, 0});
+  }
+
+  VectorPtr makeMapOfArraysVector(
+      std::map<
+          std::optional<int64_t>,
+          std::optional<std::vector<std::optional<int64_t>>>> map) {
+    std::vector<std::optional<int64_t>> keys;
+    std::vector<std::optional<std::vector<std::optional<int64_t>>>> values;
+    for (const auto& [key, vals] : map) {
+      keys.push_back(key);
+      values.push_back(vals);
+    }
+
+    auto arrayValues = vectorMaker_.arrayVectorNullable(values);
+
+    auto mapKeys = makeNullableFlatVector<int64_t>(keys);
+    auto size = mapKeys->size();
+    auto offsets = AlignedBuffer::allocate<vector_size_t>(size, pool_.get());
+    auto sizes = AlignedBuffer::allocate<vector_size_t>(size, pool_.get());
+
+    auto rawOffsets = offsets->asMutable<vector_size_t>();
+    auto rawSizes = sizes->asMutable<vector_size_t>();
+
+    // Create a vector of maps with 1 key , 1 value
+    vector_size_t offset = 0;
+    for (vector_size_t i = 0; i < size; i++) {
+      rawSizes[i] = 1;
+      rawOffsets[i] = offset;
+      offset += 1;
+    }
+
+    return std::make_shared<MapVector>(
+        pool_.get(),
+        MAP(BIGINT(), ARRAY(BIGINT())),
+        nullptr,
+        size,
+        offsets,
+        sizes,
+        mapKeys,
+        arrayValues);
+  }
+
+  std::optional<std::vector<std::optional<int64_t>>> O(
+      std::vector<std::optional<int64_t>> data) {
+    return std::make_optional(data);
+  };
+};
+
+TEST_F(PrestoHasherTest, ints) {
+  testIntegral<int8_t>();
+  testIntegral<int16_t>();
+  testIntegral<int32_t>();
+  testIntegral<int64_t>();
+  assertHash<int32_t>({99999, std::nullopt}, {-6111805494244633133, 0});
```

Comment:
The code would be more readable if there was a test per type, e.g. tinyints, smallints, integers, etc.